index.bs

```diff
@@ -2459,6 +2495,22 @@ To <dfn>parse aggregation keys</dfn> given a [=map=] |map|:
     1. [=map/Set=] |aggregationKeys|[|key|] to |keyPiece|.
 1. Return |aggregationKeys|.
 
+To <dfn>parse named budgets for source</dfn> given a [=map=] |map|:
+
+1. Let |namedBudgets| be a new [=map=].
+1. If |map|["<code>[=source-registration JSON key/named_budgets=]</code>"] does not [=map/exists|exist=], return |namedBudgets|.
+1. Let |values| be |map|["<code>[=source-registration JSON key/named_budgets=]</code>"].
+1. If |values| is not a [=map=], return an error.
+1. If |values|'s [=map/size=] is greater than the
+    [=max named budgets per source registration=], return an error.
+1. [=map/iterate|For each=] |key| → |value| of |values|:
+    1. If |key|'s [=string/length=] is greater than the [=max length per budget name for source=],
+        return an error.
+    1. If |value| is not an integer or is less than or equal to 0 or is greater than
```

Comment:
Just checking: Would there be a benefit to allowing 0 here? For example, could a source indicate that it never wants to produce a report for a particular named budget? I guess this overlaps to some extent with filtering.

I guess we can always lift this restriction later if there's a use case, so I'm fine leaving it for now.

Reply:
+1 to zero. This reminds me of the report limit, which we allow zero on. What's the reasoning not to support zero?

Reply:
I’m fine with allowing 0, but I think this can also be achieved with filtering in aggregatable_values.

Reply:
updated spec and header validator.